pandas/core/groupby/generic.py

```diff
@@ -1220,11 +1220,14 @@ def _apply_to_column_groupbys(self, func):
         return func(self)
 
     def pct_change(self, periods=1, fill_method='pad', limit=None, freq=None):
-        """Calculate percent change of each value to previous entry in group"""
-        filled = getattr(self, fill_method)(limit=limit)
-        shifted = filled.shift(periods=periods, freq=freq)
-
-        return (filled / shifted) - 1
+        """Calcuate pct_change of each value to previous entry in group"""
+        with _group_selection_context(self) as new:
+            if fill_method:
+                new = copy.copy(new)
+                new.obj = getattr(new, fill_method)(limit=limit)
+                new._reset_cache('_selected_obj')
```

Comment:
Despite being in the context manager, this line is still necessary. The context manager exit calls _reset_group_selection on line 471 of groupby/groupby.py. Here since if self._group_selection is not None does not evaluate to True, it does not make a call to reset cache.